### ISSUE TITLE
DBPTK displays datatype NUMERIC(1000,1000) differently from original data

### ISSUE THREAD

**Issue by Laurira** [keeps/dbptk-ui#265]
User has a field NUMERIC(1000,1000)
![image](https://user-images.githubusercontent.com/3479156/112978590-35c62c80-9160-11eb-985b-66aa46d56ee6.png)

It has integer values in it, some are negative, some are positive. For example (tableXX.xml):
![image](https://user-images.githubusercontent.com/3479156/112978678-4ecedd80-9160-11eb-83eb-60ec16919210.png)

Now when user uploads this siard into the DBPTK Desktop then it displays as "9.39089854E8":
![image](https://user-images.githubusercontent.com/3479156/112979055-c13fbd80-9160-11eb-9e35-b3ffcee8591c.png)

Users are expecting that data is presented exactly in the same format as original

**Comment by hmiguim** [comment]
Hi Lauri,

I'll take a lot at this issue

**Comment by hmiguim** [comment]
@Laurira is this the same database from the issue #263?

**Comment by luis100** [comment]
This issue needs to be divided into two issues:
1. Presentation options for numbers (including enabling/disabling scientific notation, setting precision decimals, grouping and currency support)
2. Support for float-pointing numbers with precision bigger than the one supported by Solr

**Comment by hmiguim** [comment]
This issue started with: https://github.com/keeps/dbptk-ui/issues/263. The problem identified involves the DBPTK Developer tool and DBPTK-UI tool. As stated in https://github.com/keeps/dbptk-developer/issues/477, PostgreSQL databases when have columns with NUMERIC data type without any precision or scale creates a column in which numeric values of any precision and scale can be stored, up to the implementation limit on precision. 

Previous to the intervention for issue https://github.com/keeps/dbptk-developer/issues/477, DBPTK Developer had the following behavior: https://github.com/keeps/dbptk-developer/blob/c8a71c2406815152c00bd195ed1ce56371e8e1e5/dbptk-modules/dbptk-module-postgresql/src/main/java/com/databasepreservation/modules/postgresql/in/PostgreSQLJDBCDatatypeImporter.java#L151-L157

If the NUMERIC datatype was declared without any precision or scale DBPTK Developer transform the column in NUMERIC. And this behavior was responsible to trigger the error found in issue https://github.com/keeps/dbptk-ui/issues/263.

DBPTK-UI when indexing the SIARD migrates data types. And following this logic since the data type doesn't have a positive scale the migration assumes that is a Long value.

https://github.com/keeps/dbptk-ui/blob/8ec4517174c83e46ce78b17f1597bb5709cb43c3/src/main/java/com/databasepreservation/common/transformers/ToolkitStructure2ViewerStructure.java#L580-L586

When indexing the data Solr tries to convert a value that as a decimal point to a value that doesn't allow any decimal point. Thus, giving the error in https://github.com/keeps/dbptk-ui/issues/263. 

The solution at that time was to remove the definition of a data type NUMERIC without any precision or scale. Reasons why:

1. It was not complying with the SIARD 2.1 specification as ID P_4.3-3 states that NUMERIC types are converted in XML decimal data type (xs:decimal). And XML decimal always has a decimal point (https://www.w3.org/TR/xmlschema11-2/#decimal). The validation of the SIARD fails if this doesn't change.

2.  PostgreSQL states that "Specifying NUMERIC without any precision or scale creates a column in which numeric values of any precision and scale can be stored, up to the implementation limit on precision." Therefore the migration between database data type to SIARD data type must comply to every numeric number that could be thrown to the column. In order to do that we must set to the max precision and scale.

And this causes the problem of visualization stated in this issue. Now, when specifying the column data type on Solr is no longer a Long but a Double. And due to the logic used by Solr (as shown in the excerpt of code below): 

```java
String value = "123456789";

System.out.println(Double.parseDouble(value));
// result: 1.23456789E8

value =  "1234";

System.out.println(Double.parseDouble(value));
// result: 1234.0
```
it adds the scientific notation or the .0. Even though the value is a Long.

In order to solve this visualization issue, we can give the user the possibility to customize how the value is displayed. As we already do with BLOB columns. Using for that GWT NumberFormat class (http://www.gwtproject.org/javadoc/latest/com/google/gwt/i18n/client/NumberFormat.html)

Another problem that may happen is the loss of precision. As shown in the example below (one example with a Long as Double and one with a Double as Double.

```java
// Long as Double
String value = "12345678900987654321";

System.out.println(Double.parseDouble(value));
// result: 1.2345678900987654E19
    
BigDecimal bd = BigDecimal.valueOf(Double.parseDouble(value));
System.out.println(bd.toPlainString());
// result: 12345678900987654000

// Double as Double
String value = "0.12345678900987654321";

System.out.println(Double.parseDouble(value));
// result: 0.12345678900987654

BigDecimal bd = BigDecimal.valueOf(Double.parseDouble(value));
System.out.println(bd.toPlainString());
// result: 0.12345678900987654
```
In summary:

- It is a good solution to convert NUMERIC without any precision or scale to NUMERIC(1000,1000) for PostgreSQL databases? Reminding that the previous conversion DBPTK Developer was doing is a violation of the SIARD specification

- The loss precision needs to be handle although with some trade-offs. We can copy the value as it is in the SIARD and index it to display but for range queries we are limited to the precision offered by Solr

- The data visualization can be enhanced to better comprehension

**Comment by luis100** [comment]
Interface options for number format based on Mongo Charts:

![image](https://user-images.githubusercontent.com/631728/117466271-6dd44280-af4a-11eb-8443-674922448472.png)
![image](https://user-images.githubusercontent.com/631728/117466329-7dec2200-af4a-11eb-90e7-eb94fd35965c.png)
